### DISCUSSION TITLE
Is it possible to install package locally in a workspace

### DISCUSSION THREAD

**Discussion by dadolyner** [vercel/turborepo#6126] [Help] [answered]
I have a problem with turborepo because i need to install a package as a local dependency that will install 5he package in a workspace node_modules and not in the root node_modules of the monorepo.

Is that even possible and if it is can someone please help me how can i achieve this.

**Comment by anthonyshew** [comment, marked as answer]
This is your package manager's job. Depending on which package manager you are using, your package manager's configuration, and your dependency tree, your dependencies may be installed in differing locations. So, altogether, out of scope for Turborepo!

Happy to help out still, though. I'm interested to know why your requirements are that the dependency *must* be installed in the package?

**Reply by 0xPT** [reply]
Hi @anthonyshew! I actually just ran into this and could use some help.

Here's the use case: One of my "apps" is an Express server that I am deploying Lambda with Terraform & Github Actions. My build process is working fine, and I'm getting an index.js file for the "app". The Express server relies on two "packages" in packages/*. The root node_modules has these, so it works fine locally. However, I need to be able to also include node_modules to include in the zip file that I upload to Lambda. I don't want to upload the entire monorepo's node_modules to the lambda, only the relevant ones to my Express server. Because the server relies on two "packages", they need to be bundled into node_modules with the rest of the server's dependencies.

It looks like [pruning](https://turbo.build/repo/docs/handbook/deploying-with-docker#the-solution) is a step in the right direction, but that doesn't put them in node_modules, it just puts them all in the same folder.

Thanks!

**Reply by k-atama** [reply]
@0xPT did you ever come up with a solution here? And by "solution" I suppose I mean "did anything click that made sense in terms of what you might have been missing process-wise?"

I'm in the same situation I think, which is I have multiple "apps" that each get deployed as individual Lambda functions using CDK. Generally I actually don't have many dependencies outside of the AWS JS SDK which is available in the Lambda runtime / path somewhere, but I do have a few dependencies that I'd like bundled. 

Anything you've learned would be great, I suspect the issue is more about how I'm thinking about turborepo.

**Reply by Nevon** [reply]
I'm in the same boat, although I agree that this is the package manager's job rather than turborepo's. We end up solving it for most lambdas by using bundling using esbuild or swc so that everything is in one file. That doesn't work for dependencies that are not bundling friendly, however (native modules, things that end up doing dynamic require calls, etc.). For those cases, we end up with some horrendous `cp ../../node_modules/some-dependency dist/node_modules/` as part of the workspace build and then marking that dependency as external in the bundler. Alternatively, we have to build lambda layers that contain only those dependencies.
